cluster-autoscaler/cloudprovider/azure/azure_vms_pool.go

```diff
@@ -170,6 +454,7 @@ func (agentPool *VMsPool) Nodes() ([]cloudprovider.Instance, error) {
 
 // TemplateNodeInfo is not implemented.
 func (agentPool *VMsPool) TemplateNodeInfo() (*schedulerframework.NodeInfo, error) {
+	// TODO(wenxuan): implement this method when vms pool can fully support GPU nodepool
```

Comment:
Not having this I think means scaling from zero is not supported (even on non-GPU nodepools)